src/preferences/upgrade.cpp

```diff
@@ -29,6 +30,61 @@ Upgrade::Upgrade()
 Upgrade::~Upgrade() {
 }
 
+namespace {
+// mapping to proactively move users to the new all-shader waveform types
+WaveformWidgetType::Type upgradeToAllShaders(WaveformWidgetType::Type waveformType) {
+    switch (waveformType) {
+    case WaveformWidgetType::EmptyWaveform:
+        return waveformType;
+    case WaveformWidgetType::SoftwareSimpleWaveform:
+        return WaveformWidgetType::AllShaderSimpleWaveform;
+    case WaveformWidgetType::SoftwareWaveform:
+        return WaveformWidgetType::AllShaderRGBWaveform;
```

Comment:
I'm not sure if we want to upgrade to an accelerated waveform since those exist purely for the cases where accelerated waveforms are not available. If acceleration is not available, mixxx should still automatically fall back to the corresponding CPU-waveform right?

Reply:
Yes, you are right. I will changes these to return waveformType.